src/flint/fmpz.jl

```diff
@@ -306,64 +302,56 @@ round(::Type{ZZRingElem}, x::ZZRingElem) = x
 
-for (fJ, fC) in ((:+, :add), (:-,:sub), (:*, :mul),
-                 (:&, :and), (:|, :or), (:xor, :xor))
-    @eval begin
-        function ($fJ)(x::ZZRingElem, y::ZZRingElem)
-            z = ZZRingElem()
-            ccall(($(string(:fmpz_, fC)), libflint), Nothing,
-                  (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, x, y)
-            return z
-        end
+for (fJ, fC) in ((:+, :add), (:-, :sub), (:*, :mul), (:&, :and), (:|, :or), (:xor, :xor))
+  @eval begin
+    function ($fJ)(x::ZZRingElem, y::ZZRingElem)
+      z = ZZRingElem()
+      ccall(($(string(:fmpz_, fC)), libflint), Nothing, (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, x, y)
+      return z
     end
+  end
 end
 
 # Metaprogram to define functions fdiv, cdiv, tdiv, div
 
-for (fJ, fC) in ((:fdiv, :fdiv_q), (:cdiv, :cdiv_q), (:tdiv, :tdiv_q),
-                 (:div, :fdiv_q))
-    @eval begin
-        function ($fJ)(x::ZZRingElem, y::ZZRingElem)
-            iszero(y) && throw(DivideError())
-            z = ZZRingElem()
-            ccall(($(string(:fmpz_, fC)), libflint), Nothing,
-                  (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, x, y)
-            return z
-        end
+for (fJ, fC) in ((:fdiv, :fdiv_q), (:cdiv, :cdiv_q), (:tdiv, :tdiv_q), (:div, :fdiv_q))
+  @eval begin
+    function ($fJ)(x::ZZRingElem, y::ZZRingElem)
+      iszero(y) && throw(DivideError())
+      z = ZZRingElem()
+      ccall(($(string(:fmpz_, fC)), libflint), Nothing, (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, x, y)
+      return z
     end
+  end
 end
 
 # N.B. we do not export the internal definition of div
 # which agrees with the internal definition of AbstractAlgebra
 # Here we set Base.div to a version that agrees with Base
 function Base.div(x::ZZRingElem, y::ZZRingElem)
-    iszero(y) && throw(DivideError())
-    z = ZZRingElem()
-    ccall((:fmpz_tdiv_q, libflint), Nothing,
-	  (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, x, y)
-    return z
+  iszero(y) && throw(DivideError())
+  z = ZZRingElem()
+  ccall((:fmpz_tdiv_q, libflint), Nothing, (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, x, y)
+  return z
 end
 
 function divexact(x::ZZRingElem, y::ZZRingElem; check::Bool=true)
-    iszero(y) && throw(DivideError())
-    z = ZZRingElem()
-    if check
-       r = ZZRingElem()
-       ccall((:fmpz_tdiv_qr, libflint), Nothing,
-             (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, r, x, y)
-       r != 0 && throw(ArgumentError("Not an exact division"))
-    else
-       ccall((:fmpz_divexact, libflint), Nothing,
-                           (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, x, y)
-    end
-    return z
+  iszero(y) && throw(DivideError())
+  z = ZZRingElem()
+  if check
+    r = ZZRingElem()
+    ccall((:fmpz_tdiv_qr, libflint), Nothing, (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, r, x, y)
```

Comment:
Without the `join_lines_based_on_source` it removes newlines where the hand-formatted code deliberately had them. I don't like it.